corehq/apps/callcenter/utils.py

```diff
@@ -3,87 +3,111 @@
 from casexml.apps.case.xml import V2
 import uuid
 from xml.etree import ElementTree
+from corehq.apps.app_manager.const import USERCASE_TYPE
 from corehq.apps.hqcase.utils import submit_case_blocks, get_case_by_domain_hq_user_id
 from couchdbkit.exceptions import MultipleResultsFound
 from corehq.elastic import es_query
+from dimagi.utils.couch import CriticalSection
 
 
-def sync_user_cases(commcare_user):
+def sync_user_case(commcare_user, case_type, owner_id, copy_user_data=True):
     """
     Each time a CommCareUser is saved this method gets called and creates or updates
     a case associated with the user with the user's details.
+
+    This is also called to create user cases when the usercase is used for the
+    first time.
     """
-    domain = commcare_user.project
-    if not (domain and domain.call_center_config.enabled):
-        return
+    with CriticalSection(['user_case_%s_for_%s' % (case_type, commcare_user._id)]):
+        domain = commcare_user.project
+
+        def valid_element_name(name):
+            try:
+                ElementTree.fromstring('<{}/>'.format(name))
+                return True
+            except ElementTree.ParseError:
+                return False
+
+        # remove any keys that aren't valid XML element names
+        fields = {k: v for k, v in commcare_user.user_data.items() if valid_element_name(k)} if copy_user_data else {}
```

Comment:
Still need to remove these from the user case (but keep them for the call center cases)